### PR TITLE
Reposition correctly element after its size was changed

### LINKED ISSUE #3743: igxOverlay should propose a way to recalculate its height, when overlay content height has changed
### Description  
igxOverlay should propose a way to recalculate its height, when overlay content height has changed

### Attachments  
The following attachment shows when drop down is filtered, then even the `reposition()` method of the `toggle` is called, the height is not recalculated and drop down is not positioned aligned to the target element.
image

### PR DESCRIPTION
**Pull request by wnvko**
We were measuring the element we should show in the overlay and was caching its size. If the size of the element was changed while element is shown on the overlay, e.g. filtering a list of items, the size of the content element (div which wraps shown element ) did not change. Now we are no longer cache the size of the element, but only its width. The height of the element gets calculated on each call to position method of the position strategy. 

Closes #3743

### Additional information (check all that apply):
 - [x] Bug fix

### Checklist:
 - [x] All relevant tags have been applied to this PR
 - [x] This PR includes unit tests covering all the new code
 - [ ] This PR includes API docs for newly added methods/properties
 - [x] This PR includes `feature/README.MD` updates for the feature docs
 - [ ] This PR includes general feature table updates in the root `README.MD`
 - [x] This PR includes `CHANGELOG.MD` updates for newly added functionality
 - [ ] This PR contains breaking changes
 - [ ] This PR includes `ng update` migrations for the breaking changes
 - [x] This PR includes behavioral changes and the feature specification has been updated with them

### CODE REVIEW

File: projects/igniteui-angular/src/lib/core/styles/components/overlay/_overlay-theme.scss [hunk 1]
```diff
@@ -79,6 +79,10 @@
         pointer-events: initial;
     }
 
+    %igx-toggle--hidden {
+        display: none !important;
```
Comment (simeonoff) [inline projects/igniteui-angular/src/lib/core/styles/components/overlay/_overlay-theme.scss]: Why do we use important here?

File: src/app/overlay/overlay.sample.ts [hunk 2]
```diff
@@ -261,10 +221,131 @@ export class OverlaySampleComponent {
...
+                this.items.push(`Item ${item}`);
+            }
+            this.cdr.detectChanges();
+            this.onChange2();
+            this._overlaySettings.positionStrategy.settings.target = this.button.nativeElement;
+        }
         this.igxDropDown.toggle(this._overlaySettings);
     }
 
+    public resizeAndSoOn() {
```
Comment (kdinev) [inline src/app/overlay/overlay.sample.ts]: Seriously?

File: projects/igniteui-angular/src/lib/core/styles/components/overlay/_overlay-theme.scss (hunk 1 above)
Comment (simeonoff) [reply projects/igniteui-angular/src/lib/core/styles/components/overlay/_overlay-theme.scss]: Ok, disregard for now.

File: src/app/overlay/overlay.sample.ts (hunk 2 above)
Comment (wnvko) [reply src/app/overlay/overlay.sample.ts]: removed
shame on me :confused:

**Pull request merged**